Brew --build-from-source

Hi there,
I read https://github.com/ledger/ledger/issues/2038 about Ledger being installed `without Python support`.
After reading the issue I tried to compile it from source completely through `brew` with the following steps :

1. Run `brew install boost-python3`
2. Run `brew edit ledger` to edit the `ledger.rb` Homebrew formula
3. Add `--python` to `args` at line 46, save file
4. Run `export HOMEBREW_NO_INSTALL_FROM_API=1` to tell `brew` to use the edited local formula
5. Run `sudo chown -R $(whoami) $(brew --prefix)/*` to ensure `brew` has the right read/write permissions
6. Run `brew reinstall ledger --build-from-source`
7. Encounter the following compile error :

```
Warning: building from source is not supported!
You're on your own. Failures are expected so don't create any issues, please!
==> Fetching ledger
==> Downloading https://github.com/ledger/ledger/archive/v3.3.2.tar.gz
Already downloaded: /Users/User/Library/Caches/Homebrew/downloads/0f24302e52c7d167be5d610631e0562d4ecde4e1afa53a12cf3f48c4d65ccf0b--ledger-3.3.2.tar.gz
==> Reinstalling ledger
==> ./acprep opt make --jobs=4 --output=build --python --prefix=/usr/local/Cellar/ledger/3.3.2_1 --boost=/usr/local/opt/boost -- -DBUILD_DOCS=1 -DBUILD_WEB_DOCS=1 -DBoost_NO_BOOST
==> ./acprep opt make doc --jobs=4 --output=build --python --prefix=/usr/local/Cellar/ledger/3.3.2_1 --boost=/usr/local/opt/boost -- -DBUILD_DOCS=1 -DBUILD_WEB_DOCS=1 -DBoost_NO_B
==> ./acprep opt make install --jobs=4 --output=build --python --prefix=/usr/local/Cellar/ledger/3.3.2_1 --boost=/usr/local/opt/boost -- -DBUILD_DOCS=1 -DBUILD_WEB_DOCS=1 -DBoost_
Last 15 lines from /Users/User/Library/Logs/Homebrew/ledger/03.acprep:
-- Installing: /usr/local/Cellar/ledger/3.3.2_1/include/ledger/views.h
-- Installing: /usr/local/Cellar/ledger/3.3.2_1/include/ledger/xact.h
-- Installing: /usr/local/Cellar/ledger/3.3.2_1/include/ledger/system.hh
-- Installing: /usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ledger.so
CMake Error at src/cmake_install.cmake:144 (file):
  file INSTALL cannot copy file
  "/tmp/ledger-20230630-26352-1rknm8z/ledger-3.3.2/build/ledger.so" to
  "/usr/local/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/ledger.so":
  Operation not permitted.
Call Stack (most recent call first):
  cmake_install.cmake:42 (include)


make: *** [install] Error 1
acprep: ERROR: Execution failed: make install -Wno-dev -j4

READ THIS: https://docs.brew.sh/Troubleshooting
```

Maybe this is linked to this issue as well https://github.com/ledger/ledger/issues/531.
Not sure either if I should post on https://github.com/Homebrew/brew/ issue, because of the `Failures are expected so don't create any issues, please!`

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -244,6 +244,12 @@ if(USE_PYTHON)
       set(_ledger_python_module_name "ledger${CMAKE_SHARED_LIBRARY_SUFFIX}")
     endif()
 
+    # Allow packagers to override where the Python module is installed.
+    # By default this uses Python's site-packages, but systems like Homebrew
+    # may need to install under CMAKE_INSTALL_PREFIX instead.
+    set(LEDGER_PYTHON_INSTALL_DIR "${Python_SITEARCH}" CACHE PATH
+      "Directory to install the ledger Python module into")
+
     # FIXME: symlink would be sufficient:
     # maybe using install(CODE "...") and
     # execute_process(COMMAND "${CMAKE_COMMAND}" -E create_symlink ...).
@@ -254,7 +260,7 @@ if(USE_PYTHON)
       $<TARGET_FILE:libledger> "${CMAKE_BINARY_DIR}/${_ledger_python_module_name}")
     install(
       FILES "${CMAKE_BINARY_DIR}/${_ledger_python_module_name}"
-      DESTINATION ${Python_SITEARCH})
+      DESTINATION "${LEDGER_PYTHON_INSTALL_DIR}")
   else()
     message(WARNING "Python_SITEARCH not set. Will not install python module.")
   endif()

